Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,453 @@
+    , DeviceCustomNumber1Label = column_ifexists("DeviceCustomNumber1Label", "") // DeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber2Label = column_ifexists("DeviceCustomNumber2Label", "") // DeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber3Label = column_ifexists("DeviceCustomNumber3Label", "") // Connection Duration in milliseconds (for Connnection Closed msgs)
+    , fw_action = column_ifexists("fw_action", "")
+    , sosFirewallAction = column_ifexists("fw_action", "")
+    , susr = column_ifexists("susr", "") // Logged-in username associated with the log event.
+    , af_polid = column_ifexists("af_polid", "")
+    , af_action = column_ifexists("af_action", "")
+    , af_policy = column_ifexists("af_policy", "")
+    , af_service = column_ifexists("af_service", "")
+    , af_type = column_ifexists("af_type", "")
+    , af_object = column_ifexists("af_object", "")
+    , contentObject = column_ifexists("contentObject", "") // App Firewall/App Rule Content.
+    , srcV6 = column_ifexists("srcV6", "") // Source IPv6 IP
+    , dstV6 = column_ifexists("dstV6", "") // Destination IPv6 IP
+    , appName = column_ifexists("appName", "") // Non-signature Application name that matches the Application ID "app" or "f".
+    , app = column_ifexists("app", "") // Numeric Application ID. Not the same as "ApplicationProtocol".
+    , sosAppCategoryName = extract(@'^["]([a-zA-Z-]+)', 1, column_ifexists("appcat", "")) // Application Category from App Control
+    , sosAppIDName = extract(@' (.*) --', 1, column_ifexists("appcat", "")) // Application Name from App Control
+    , sosAppSignatureName = trim('"', extract(@'.*-- (.*\S)', 1, column_ifexists("appcat", ""))) // Application Signature Name from App Control
+    , catid = column_ifexists("catid", "") // Application Category ID
+    , sid = column_ifexists("sid", "") // Application Signature ID
+    , appid = column_ifexists("appid", "") // Application ID from App Control
+    , sosIPSFullString = column_ifexists("ipscat", "")
+    , ipscat = extract(@'^"?([a-zA-Z-\/]+)', 1, column_ifexists("ipscat", "")) // IPS Category/Signature
+    , sosIPSSignatureName = extract(@'[ ](.*)\S', 1, column_ifexists("ipscat", "")) // IPS Signature name
+    , spycat = column_ifexists("spycat", "") // // Anti-Spyware Category.
+    , FileSize = tolong(column_ifexists("FileSize", long(null)))
+    , FileName = column_ifexists("FileName", "")
+    , sosAppControlFileName = extract(@'.*Filename: (.*)\"', 1, column_ifexists("Message", "")) // App Control Filename Logging
+    , sosHttpReferrer = extract(@'Referer: (.*)\"$', 1, column_ifexists("sosLogMsgNote", ""))
+    , sosHttpRequestMethod_ = extract(@'Command: (.\w+)', 1, column_ifexists("sosLogMsgNote", ""))
+    , sosCaptureATPVerdict = extract(@'Gateway Anti-Virus Status: (.*)\. ', 1, column_ifexists("Message", ""))
+    , sosCountry = extract(@'Country Name:(.*)\"$', 1, column_ifexists("Message", ""))

```

Comment:
we have DstGeoCountry or SrcGeoCountry in network schema. this value not fitting in these fields?

Reply:
Got it, thanks